src/components/Dropdown/Dropdown.tsx

```diff
@@ -162,20 +172,33 @@ function useFocusContextInstance(
   lastTypedOrSubmittedValue: string,
   setValue: (newValue: string) => void,
   screenReaderKey: number,
-  setScreenReaderKey: (newKey: number) => void
+  setScreenReaderKey: (newKey: number) => void,
+  alwaysSelectOption: boolean
 ): FocusContextType {
   const [focusedIndex, setFocusedIndex] = useState(-1);
   const [focusedValue, setFocusedValue] = useState<string | null>(null);
   const [focusedItemData, setFocusedItemData] = useState<Record<string, unknown> | undefined>(undefined);
 
+  useEffect(() => {
+    if (alwaysSelectOption && items.length > 0) {
+      setFocusedIndex(0);
+      setFocusedItemData(items[0].itemData);
+    }
+  }, [alwaysSelectOption, items]);
+
   function updateFocusedItem(updatedFocusedIndex: number, value?: string) {
     const numItems = items.length;
     let updatedValue;
     if (updatedFocusedIndex === -1 || updatedFocusedIndex >= numItems || numItems === 0) {
       updatedValue = value ?? lastTypedOrSubmittedValue;
-      setFocusedIndex(-1);
-      setFocusedItemData(undefined);
-      setScreenReaderKey(screenReaderKey + 1);
+      if (alwaysSelectOption && numItems !== 0) {
```

Comment:
does this prevent us from arrowing up to loop backwards through the dropdown? I know that's the desired behavior for the search bar, but not sure if that's an issue for filter search or not

Reply:
Not sure if you're already working on this, but it looks like both of these are still a problem

Reply:
I tested this out again and the looping/tabbing works! The only thing now is that when a user types something, arrows/tabs through the dropdown and then back to the input, and then clicks outside, the first filter in the dropdown is no longer selected and applied on the next search